### PR TITLE
Running validator against corpus

### PR DESCRIPTION
**Pull request by dockmd**
Added validate-all-testcases file that runs scripts/validate-testcases.py for each testcase.xml in the repository.
Tasks of the script:

- checks if testcase.xml can be parsed,
- checks if testcase.xml should be validated,
- checks if testcase.xml has the necessary content to perform validation,
- performs the validation,
- tries to match the output from the validator with expected value from testcase.xml. 

Validation is also performed every time a new commit is pushed to the integration branch.
Example result: https://github.com/DILCISBoard/eark-ip-test-corpus/actions/runs/8175006698/job/22351143598

@PhillipAasvangTommerholt @carlwilson 
Please take a look, as none of the 435 validations gave a result that satisfied the expected result.

### CODE REVIEW

**Review by carlwilson — DISMISSED** [review]
Looks a solid start.

File: .github/workflows/validate-testcases.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Validating test cases
+        run: |
+          python -m pip install --upgrade pip
+          cd validator
+          pip install .
+          cd ../corpus
+          chmod +rx ./validate-all-testcases
```
Comment (carlwilson) [inline .github/workflows/validate-testcases.yaml]: This line seems to misfire in real life.

**Comment by carlwilson** [comment]
Although there's an error in the execution it appears? Something to do with the `cd corpus` followed by the `chmod` (and probably the execution on the following workflow line).

**Comment by dockmd** [comment]
> Although there's an error in the execution it appears? Something to do with the `cd corpus` followed by the `chmod` (and probably the execution on the following workflow line).

It failed because the validate-all-testcases file was not yet in the integration branch. I will add workflow in the new PR after this one is completed.

File: .github/workflows/validate-testcases.yaml (hunk 1 above)
Comment (dockmd) [reply .github/workflows/validate-testcases.yaml]: I will add the workflow in the next PR with minor changes

**Review by carlwilson — APPROVED** [review]
This looks good to me and in truth, the proof of the pudding is when it's running I guess. This is wired up to a GH action somewhere I guess.

**Pull request merged**